crates/primitives/src/eip7702/bytecode.rs

```diff
@@ -0,0 +1,102 @@
+use crate::{bytes, Address, Bytes};
+
+/// EIP-7702 Version Magic in u16 form.
+pub const EIP7702_MAGIC: u16 = 0xEF01;
+
+/// EIP-7702 magic number in array form.
+pub static EIP7702_MAGIC_BYTES: Bytes = bytes!("ef01");
+
+/// EIP-7702 first version of bytecode.
+pub const EIP7702_VERSION: u8 = 0;
+
+/// Bytecode of delegated account, specified in EIP-7702
+///
+/// Format of EIP-7702 bytecode consist of:
+/// 0xEF00 (MAGIC) + 0x00 (VERSION) + 20 bytes of address.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Eip7702Bytecode {
+    pub delegated_address: Address,
+    pub version: u8,
+    pub raw: Bytes,
+}
+
+impl Eip7702Bytecode {
+    /// Creates a new EIP-7702 bytecode or returns None if the raw bytecode is invalid.
+    #[inline]
+    pub fn new(raw: Bytes) -> Option<Self> {
```

Comment:
nit: prob better to have `new` take address and do a `new_raw` instead (i.e. flip `new` and `new_address`)

Reply:
It is better to switch them.